projects/03/a/RAM64.hdl

```diff
@@ -15,5 +15,22 @@ CHIP RAM64 {
     OUT out[16];
 
     PARTS:
-    // Put your code here:
-}
\ No newline at end of file
+
+    // Decide which ram to load
+    DMux8Way(in=load, sel=address[0..2], a=load1, b=load2, c=load3,
+    d=load4, e=load5, f=load6, g=load7, h=load8);
+
+    // Load ram according to address
+    RAM8(in=in, load=load1, address=address[3..5], out=ram1);
+    RAM8(in=in, load=load2, address=address[3..5], out=ram2);
+    RAM8(in=in, load=load3, address=address[3..5], out=ram3);
+    RAM8(in=in, load=load4, address=address[3..5], out=ram4);
+    RAM8(in=in, load=load5, address=address[3..5], out=ram5);
+    RAM8(in=in, load=load6, address=address[3..5], out=ram6);
+    RAM8(in=in, load=load7, address=address[3..5], out=ram7);
+    RAM8(in=in, load=load8, address=address[3..5], out=ram8);
+
+    // Return the output of whichever ram it was
+    Mux8Way16(a=ram1, b=ram2, c=ram3, d=ram4,
+    e=ram5, f=ram6, g=ram7, h=ram8, sel=address[0..2], out=out);

```

Comment:
Another extremely minor naming quibble: the internal pins are numbered `…1`, `…2`, `…3` etc, but the `DMux8Way` and `Mux8Way16` pins are lettered `a`, `b`, `c` etc. Is it important for these to be different or could we be consistent throughout to avoid errors/confusion?

(This comment applies to all the `RAM*.hdl` chips I guess.)

Reply:
In retrospect maybe naming them `ramA`, `ramB`, `ramC` might make things clearer, but I'm too lazy to go back to fix the RAM*.hdl chips. Minor naming quibble noted for next time!

Reply:
😂 👍🏻